### DISCUSSION TITLE
(DONE) Update of Jupiter IMCCE/BDL data-files and according code + openSSL

### DISCUSSION THREAD

**Discussion by obspsr** [XEphem/XEphem#19] [Ideas]
Hi Brandon,
As I told you months ago, I managed with Valery Lainey astronomer in charge of satellites files to update. 
We finished today check of right format for jupiter and moons, the other 4 planets will follow soon this week ...
- It works as usual (graphics and info panel) extending use to 2040.

- I applied the 2 SSL related patches from Lutz Mändle < lmaendle at gmx dot net > of 2016 and 2018 which give 
access to Data -> Download https:// databases and SUN/SOHO images.(Needs openSSL library installed).

- Could you merge from my commits from (https://github.com/obspsr/XEphem.git) 
  as I have no authorization  to submit git commit to you ...!? 
(I had to fork 4.0.2) 
On Linux Mageia-8 my cloned git repos compiles and works with no heck.   

Thanks for coop and best regards,
Serge.

**Comment by brandon-rhodes** [comment]
> I have no authorization to submit git commit to you ...!?

We should get that fixed! Once you forked the project, and made commits of your own to your own, GitHub did not offer you a button to create a Pull Request? Their instructions suggest that it should be possible:

https://docs.github.com/en/github/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork

“Anyone with write access to a repository can create a pull request from a user-owned fork.”

I should be able to look over your commits tomorrow.

**Reply by obspsr** [reply]
I am really sorry, but I did not see the Pull proposition first !
It's done now.
I have a friend who should try the candidate on **Debian**, I'll call him
tomorrow. I will pull  on my new Lenovo **Ubuntu** .

**Reply by obspsr** [reply]
Hi Brandon once more,
We tried the git clone on Debian Buster, and Mageia-8, followed compilation without a heck ! (Normal with Elwood code !)
It was a pleasure to do the job with Valéry Lainey and IMCCE/BdL people !
Enjoy with all this moons next 20 years !
Thanks for coop,
Serge.

**Reply by brandon-rhodes** [reply]
Thank you for checking on more than one platform! I will hopefully have time to review and merge the pull request later this week, or by Monday if the next few days are too busy.

**Reply by obspsr** [reply]
I am surprised that all good code of last pull request as disappeared and XEphem/xephem site says "no pull request" !!??
Serge.

**Reply by brandon-rhodes** [reply]
If you click on the words “6 Closed”, you will see your pull requests listed — it looks like you closed both of them last week? Maybe you clicked the “Close” button by accident?

**Reply by obspsr** [reply]
It's completely crazy, message says you submit 9 files ! I see only Uranus pull request.
I called for about 13 files (included the ssl patches) and no any update appears ! Do i have to pull for all 4 planets and 4 BdL files also ? 
god ....

**Reply by obspsr** [reply]
I confirm that I am unable to place a pull request for the jupiter.20240 file from my fork. :((
If you have an idea ...
Serge.

**Reply by brandon-rhodes** [reply]
1. ~~Do you have a copy of the file that works with XEphem?~~
2. ~~Can you simply attach the file to a comment here, by clicking "Attach files…" at the bottom of the comment field as you type? That would let me get a copy of the file, if you have run into problems with "git" and cannot submit the file through a PR.~~

(Edit: see my next comment, below.)

**Reply by brandon-rhodes** [reply]
Oh — it looks like your Pull Request just updated, and the file looks different now, and all the data lines look like they are exactly 541 characters! When I get a chance tomorrow I will try it out with XEphem and see if it avoids the error I was getting. Thanks!

**Reply by obspsr** [reply]
Thanks ... my other friend tester forgot to tell me about the EOF warning !!! : (((
Other files are ok.

**Reply by obspsr** [reply]
Hi Brandon,
Could you confirm that you got them and that will push the 4 galilean files and code ?
We are not in a hurry, and i can use my fork but I prefer to operate from the now
official repository....
Best regards
Serge.

**Reply by brandon-rhodes** [reply]
Yes, I now see the correctly formatted file. Probably tomorrow I will have time to sit down and extract the commits for each file from the SSL merge that's also mixed into your pull request — I'll comment back here when I think the files are in place on `main`.

**Reply by obspsr** [reply]
Thanks Brandon :+1:

**Reply by brandon-rhodes** [reply]
As noted over on the pull request https://github.com/XEphem/XEphem/pull/21 the new data files are now available on the `main` branch of the project. Thanks for providing them!
